app/src/main/java/com/capstone/weatherapp/CityListViewModel.kt

```diff
@@ -1,26 +1,45 @@
 package com.capstone.weatherapp
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
 
 private const val CITIES = "4772354,5368361,5128581,4887398,4699066,5809844,5856195,5419384,5391959,4684888,5746545,4990729,4335045"
 private const val APIKEY = "6849ae760f417fb8188f4bb7fd0d92fc"
 private const val UNITS = "imperial"
 
-class CityListViewModel : ViewModel() {
-    private val _cityList = MutableLiveData<List<City>>()
+class CityListViewModel(private val weatherRepo: WeatherRepository) : ViewModel() {
+    private val _cityList = weatherRepo.cityListData
     val cityList: MutableLiveData<List<City>>
         get() = _cityList
 
-    fun getCities() {
+    private val _eventNetworkError = MutableLiveData<Boolean>(false)
+    val eventNetworkError: LiveData<Boolean>
+        get() = _eventNetworkError
+
+    fun getCityListFromRepo() {
         viewModelScope.launch {
             try {
-                _cityList.value = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS).list
+                weatherRepo.refreshCityList()
+                _eventNetworkError.value = false
             } catch (e: Exception) {
-                _cityList.value = emptyList()
+                if (cityList.value.isNullOrEmpty()) {
+                    _eventNetworkError.value = true
+                }
             }
         }
     }
+}
+
+class CityListViewModelFactory(private val repo: WeatherRepository) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(CityListViewModelFactory::class.java)) {
```

Comment:
this parameter shouldn't be the `CityListViewModelFactory::class.java` class, it should just be the viewModel (so `CityListViewModel::class.java`